docs/source/architecture.rst

```diff
@@ -1,17 +1,142 @@
+  algorithm handles initialization, normal mode (block processing), and view
+  changes
+
+.. _network-overview-label:
+
+Network Overview
+================
+
+Fault Tolerance
+---------------
+
+A PBFT network consists of nodes that are ordered from 0 to `n-1`, where
+`n` is the total number of nodes in the network.
+
+The PBFT algorithm guarantees network `safety
+<https://en.wikipedia.org/wiki/Liveness#Liveness_and_safety>`__
+as long as the number of faulty nodes remains below the required percentage.
+The maximum number of faulty nodes that the network can tolerate is determined
+by the formula :math:`f = \frac{n - 1}{3}`. In other words, no more than one
+third of the nodes (rounded down) can be out of order or dishonest at any given
+time.
+
+For example, a four-node network (the minimum required number of nodes)
+can tolerate one faulty node. Increasing the size of the network reduces the
+likelihood that that all :math:`\frac{n - 1}{3}` nodes would be faulty at the
+same time.
+
+Network `liveness
+<https://en.wikipedia.org/wiki/Liveness#Liveness_and_safety>`__
+is guaranteed as long as there are less than :math:`2f+1` unreachable nodes,
+counting both node and network failures. That is, at least third of the nodes
+(rounded up) must be reachable at all times.
+
+View Changes: Choosing a New Primary
+------------------------------------
+
+A PBFT network has one leader, called the `primary node`. All other nodes on the
+network are called `secondary nodes`.
+
+* The primary requests new blocks from the validator, notifies the network when
+  a new block is available, and reports consensus results (commit or fail) back
+  to the validator.
+
+* The secondary nodes monitor the health of the primary and request a change
+  if the primary appears out of order or malicious.
+
+A `view` is the period of time that a given node is the primary. The next
+primary is selected in a round-robin (circular) fashion, according to the order
+of nodes listed in the ``sawtooth.consensus.pbft.peers``
+:doc:`on-chain setting <on-chain-setting>`.
+In a four-node network, for example, node 0 is the primary at view 0,
+node 1 is the primary at view 1, and so on.  When the network gets to view 4,
+it will "wrap back around" so that node 0 is the primary again.
+
+The algorithm uses the formula `p = v mod n` to determine the next
+primary. In this formula, `p` is the primary, `v` is the view number, and `n` is
+the total number of nodes in the network. For example, if a four-node network is
+at view 7, the formula `7 mod 4` determines that node 3 is the primary.
+
+The Sawtooth PBFT algorithm changes the primary at regular intervals, as well as
+when the secondary nodes determine that the current primary is faulty.
+See :ref:`view-changing-mode-label` for a description of this process.
+
+Sequence Numbers
+----------------
+
+The primary assigns a `sequence number` for each new block (called a "candidate
```

Comment:
The primary doesn't really *assign* a sequence number in Sawtooth PBFT (unlike traditional PBFT), because the sequence number is already defined as the block number.

Reply:
I think this sentence (which was in the original doc) is referring to the general PBFT algorithm. I propose rewording it thusly:

The PBFT algorithm specifies that each block has a `sequence number`. Sawtooth PBFT uses the block number as the sequence number.